bots/pr/src/main/java/org/openjdk/skara/bots/pr/CheckWorkItem.java

```diff
@@ -251,6 +251,17 @@ public Collection<WorkItem> prRun(Path scratchPath) {
         var hostedRepositoryPool = new HostedRepositoryPool(seedPath);
         CensusInstance census;
         var comments = prComments();
+
+        // If merge pr is not allowed, reply warning to the user and return
+        var mergeDisabledText = "<!-- merge error -->\n" +
+                ":warning: @" + pr.author().username() + " merge PR is not allowed in this repository, please close this pr." +
+                " If it was unintentional, please modify the title of this PR.";
+        if (!bot.enableMerge() && PullRequestUtils.isMerge(pr)) {
+            addErrorComment(mergeDisabledText, comments);
+            return List.of();
+        }
+        removeErrorComment(mergeDisabledText, comments);
```

Comment:
I thought if we put the check on the top of the method, it would help us save some rest api calls. I am also ok with moving it inside the currentCheckValid block.

Reply:
That is true, we would. However, since this is such a rare code path, it's not really worth optimizing for. I think it's more important that the code is more readable.